consensus/misc/eip1559/eip1559.go

```diff
@@ -60,7 +60,11 @@ func VerifyEIP1559Header(config *params.ChainConfig, parent, header *types.Heade
 func CalcBaseFee(config *params.ChainConfig, parent *types.Header, time uint64) *big.Int {
 	// If this is the cel2 transition block and the parent block has a base fee
 	// then use that.
-	if config.Cel2Time != nil && *config.Cel2Time == time && parent.BaseFee != nil {
+	if !config.IsCel2(parent.Time) && config.IsCel2(time) && parent.BaseFee != nil {
```

Comment:
I still think this wouldn't be triggered on any migrated chain. In the migration we set the Cel2 flag on the migration block, so every parent will have it enabled.

Besides that, do we need the change here? I think the change in `CalcBaseFee` should be enough.

Reply:
So that if is never going to be reached?

What I'm doing with that if, it's required to "normalize" the baseFee, if for some reason we have it under the floor, if the baseFee increases or it's the same, will be under the floor until one block tries to reduce the baseFee.

That would happen only if the baseFee is under the Floor, which shouldn't happen because we are going to set it anyways and we should set the same Floor that the existing network has. So, for example, if the new floor is 5, and the baseFee is in 3 in the cel2 switch, without that correction,  if it keeps using the targetDensity amount of gas, will be in 3 instead of 5 as long as it hits less than the target density.

The other possibility, is to add a check in the equals and when increases, but I think that we can avoid those checks if we correct it in the block switch (and again, the only possible scenario is if we set a Floor smaller than the Floor that the testnet has)

Reply:
i don't get why we are checking 'time' here. shouldn't we just check a "feature flag" here?

like `if config.baseFeeFloor >0` then...
a call it a day?

and then add a `Config.BaseFeeFloor` option

Reply:
Refactored the code to avoid any modification on what we already had, and just check the return value against a feature flag